File: solutions/0168. Excel Sheet Column Title/0168.cpp
// Time complexity: O(logn)
// Space complexity: O(logn)
class Solution {
public:
    string convertToTitle(int columnNumber) {
        string ans;
        while (columnNumber--) {
            ans.push_back(columnNumber % 26 + 'A');
            columnNumber /= 26;
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: solutions/0168. Excel Sheet Column Title/0168.cs
// Time complexity: O(logn)
// Space complexity: O(logn)
public class Solution {
    public string ConvertToTitle(int columnNumber) {
        StringBuilder ans = new StringBuilder();
        while (columnNumber > 0) {
            columnNumber--;
            ans.Insert(0, (char)('A' + columnNumber % 26));
            columnNumber /= 26;
        }
        return ans.ToString();
    }
}